src/Workspaces/CSharp/Portable/CodeGeneration/ExpressionGenerator.cs

```diff
@@ -311,5 +312,57 @@ private static ExpressionSyntax GenerateMemberAccess(params string[] names)
+        {
+            if (canUseFieldReference || value > sbyte.MinValue)
+            {
+                return GenerateLiteralExpression(type, value, LiteralSpecialValues.SByteSpecialValues, null, canUseFieldReference, (s, v) => SyntaxFactory.Literal(s, v), x => x < 0, x => (sbyte)-x);
+            }
+
+            // We have to special case how sbyte.MinValue is printed when we can't refer to the
+            // field directly.
+            var numericLiteralToken = SyntaxFactory.Literal("-128", value);
+            return SyntaxFactory.LiteralExpression(SyntaxKind.NumericLiteralExpression, numericLiteralToken);
+        }
+
+        private static ExpressionSyntax GenerateShortLiteralExpression(ITypeSymbol type, short value, bool canUseFieldReference)
+        {
+            if (canUseFieldReference || value > short.MinValue)
+            {
+                return GenerateLiteralExpression(type, value, LiteralSpecialValues.Int16SpecialValues, null, canUseFieldReference, (s, v) => SyntaxFactory.Literal(s, v), x => x < 0, x => (short)-x);
+            }
+
+            // We have to special case how short.MinValue is printed when we can't refer to the
+            // field directly.
+            var numericLiteralToken = SyntaxFactory.Literal("-32768", value);
+            return SyntaxFactory.LiteralExpression(SyntaxKind.NumericLiteralExpression, numericLiteralToken);
+        }
+
+        private static ExpressionSyntax GenerateIntLiteralExpression(ITypeSymbol type, int value, bool canUseFieldReference)
+        {
+            if (canUseFieldReference || value > int.MinValue)
+            {
+                return GenerateLiteralExpression(type, value, LiteralSpecialValues.Int32SpecialValues, null, canUseFieldReference, SyntaxFactory.Literal, x => x < 0, x => -x);
+            }
+
+            // We have to special case how int.MinValue is printed when we can't refer to the
+            // field directly.
+            var numericLiteralToken = SyntaxFactory.Literal("-2147483648", value);
+            return SyntaxFactory.LiteralExpression(SyntaxKind.NumericLiteralExpression, numericLiteralToken);
+        }
+
+        private static ExpressionSyntax GenerateLongLiteralExpression(ITypeSymbol type, long value, bool canUseFieldReference)
+        {
+            if (canUseFieldReference || value > long.MinValue)
+            {
+                return GenerateLiteralExpression(type, value, LiteralSpecialValues.Int64SpecialValues, null, canUseFieldReference, SyntaxFactory.Literal, x => x < 0, x => -x);
+            }
+
+            // We have to special case how long.MinValue is printed when we can't refer to the
+            // field directly.
+            var numericLiteralToken = SyntaxFactory.Literal("-9223372036854775808", IntegerUtilities.ToUInt64(value));
+            return SyntaxFactory.LiteralExpression(SyntaxKind.NumericLiteralExpression, numericLiteralToken);
```

Comment:
all these functions seem nearly identical.  Can we not just pass in these values to the GenerateLiteralExpression helper instead?

Reply:
Makes sense. Updated